tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -171,6 +178,40 @@ private static void AddSamplingPriorityTags(Span span, int samplingPriority)
             }
         }
 
+        // Based on the Go implementation. Cross-check with Java
+        //
+        // Keep the entire trace when StatsComputation is not enabled
+        // When StatsComputation is enabled, perform the following checks (in the given order) when each span is finished. If any span returns true, keep the entire trace.
+        // - Is the current sampling priority > 0? Return true.
+        // - Have any errors been seen? Return true
+        // - If the current span doesn't have Metrics["_dd1.sr.eausr"] (aka AnalyticsSampleRate), skip. If it does, run the Knuth sampling decision on the metric value and return the result.
+        private bool ShouldKeepSpan(Span span)
```

Comment:
So we currently calculate whether we should keep the span here based on the rules about, then we add the extra information from the stats aggregation later too. It seems a little off to be calculating two halves of the same thing in two different places, any chance we can merge them somehow, either pushing the stats aggregation up to here, or the span inspection down? 🤔 

EDIT: I see Pierre had the same thoughts 🙂